backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CpuPinningPolicyUnit.java

```diff
@@ -67,73 +67,70 @@ public List<VDS> filter(final SchedulingContext context,
+                    messages.addMessage(host.getId(), EngineMessage.VAR__DETAIL__NO_HOST_CPU_DATA.name());
+                    log.debug("Host {} does not have the cpu topology data.", host.getId());
+                    continue;
                 }
-                break;
-            case DEDICATED:
-                for (final VDS host : hosts) {
-                    if (host.getCpuTopology() == null || host.getCpuTopology().isEmpty()) {
-                        // means CPU topology not reported for this host, lets ignore it
-                        messages.addMessage(host.getId(), EngineMessage.VAR__DETAIL__NO_HOST_CPU_DATA.name());
-                        log.debug("Host {} does not have the cpu topology data.", host.getId());
-                        continue;
-                    }
 
-                    Map<Guid, List<VdsCpuUnit>> vmToPendingDedicatedCpuPinnings =
-                            PendingCpuPinning.collectForHost(getPendingResourceManager(), host.getId());
-                    boolean isDedicatedCpuPinningPossibleAtHost =
-                            vdsCpuUnitPinningHelper.isDedicatedCpuPinningPossibleAtHost(
-                                    vmToPendingDedicatedCpuPinnings,
-                                    vm,
-                                    host.getId());
-                    if (isDedicatedCpuPinningPossibleAtHost) {
-                        candidates.add(host);
-                    } else {
-                        messages.addMessage(host.getId(), EngineMessage.VAR__DETAIL__VM_PINNING_DEDICATED_NOT_FIT.name());
-                        log.debug("Host {} does not satisfy CPU pinning constraints, cannot match virtual topology " +
-                                "with available CPUs.", host.getId());
-                    }
+                Map<Guid, List<VdsCpuUnit>> vmToPendingDedicatedCpuPinnings =
+                        PendingCpuPinning.collectForHost(getPendingResourceManager(), host.getId());
+                boolean isDedicatedCpuPinningPossibleAtHost =
+                        vdsCpuUnitPinningHelper.isExclusiveCpuPinningPossibleOnHost(
+                                vmToPendingDedicatedCpuPinnings,
+                                vm,
+                                host.getId());
+                if (isDedicatedCpuPinningPossibleAtHost) {
+                    candidates.add(host);
+                } else {
+                    messages.addMessage(host.getId(), EngineMessage.VAR__DETAIL__VM_PINNING_DEDICATED_NOT_FIT.name());
+                    log.debug("Host {} does not satisfy CPU pinning constraints, cannot match virtual topology " +
+                            "with available CPUs.", host.getId());
                 }
-                break;
+            }
+            return candidates;
+        }
+        // shared CPUs
+        if (vm.getCpuPinningPolicy() == CpuPinningPolicy.MANUAL) {
+            if (StringUtils.isEmpty(cpuPinning)) {
+                return hosts;
```

Comment:
we really need a comment here - it's easy to forget why we do that.. that's because for resize-and-pin we determine the pinning later on, right?

Reply:
Looks like with all the changes https://github.com/oVirt/ovirt-engine/commit/bf91cba3037b5a3c4cb326873a49e12a1ac48e13 gone somewhere, i'll re-add the comment and change it to `!vm.getCpuPinningPolicy().isExclusive()`

Reply:
OK, it's needed.